### ISSUE TITLE
[Bug] Favourites no longer function after the latest update.

### ISSUE THREAD

**Issue by rizreflects** [jmooremcc/Repository.jmooremcc#1]
Since the latest 2.0.9 update, the favourites I created no longer open when clicked. The only way to resolve this was to revert back to 2.0.7 and then disallow the addon from self-updating within Kodi. This is not too big of a task on my PC, as the process of reverting back is straight forward. However, I have also set up super favourites on friends/ family devices running core/libreELEC and the process of reverting back is not as straight forward on there. Setting up the superfavourites from scratch would take me hours and I simply do not have that kind of time anymore. 

As you can see in the images I have attached, a lot of time went into setting up these super favourites. I manually downloaded thumbnails and created  many of my own self-updating trakt lists from scratch. 

![superfavsdirectorsandstudios](https://user-images.githubusercontent.com/76912707/120810510-060e1900-c543-11eb-9194-0620c9581fc6.PNG)
![superfavgoodmovie](https://user-images.githubusercontent.com/76912707/120810517-07d7dc80-c543-11eb-825c-cffff93a8b2f.PNG)
![superfavdiscover](https://user-images.githubusercontent.com/76912707/120810524-08707300-c543-11eb-9a73-2b94e35548e4.PNG)
![superfavdocsandspecials](https://user-images.githubusercontent.com/76912707/120810528-0a3a3680-c543-11eb-9cba-efd367f96c5c.PNG)

If you need a log file please let me know and I will attach one. However, this is happening on 4 seperate devices so I doubt it has anything to do with my particular setup. Hopefully this problem can be resolved as this addon is as important as autowidget IMO. 

Thanks a lot!

**Comment by jmooremcc** [comment]
Thanks for the heads up.
Version 2.09 came about because Kodi Matrix 19.1 had changes that created problems in SF 2.07.
If you can create a Debug log that illustrates the problem, that would be helpful.
Based on what you've reported, SF 2.07 is the best fit for Matrix 19.0 for now.

**Comment by rizreflects** [comment]
Thanks for the response. Here is my log file https://paste.kodi.tv/oyusivitus (I'm assuming this is what you meant by debug log?)

I started Kodi and attempted to open the superfavourite I created. It loaded for about a second but never opened. The same thing happened on my second attempt. The log was uploaded right after this. 

I should have also mentioned that I'm running Kodi 19.1. Yesterday I tried to create a superfavourite from scratch on version 2.09, but it didn't open (same problem I stated in the last paragraph). 

To sum up, 2.07 works fine.. you are able to create favourites and also view what's inside. However, 2.09 seems to be broken. The favourites can be created but not actually opened.  

If there's anything I can do to help just let me know.

**Comment by jmooremcc** [comment]
I've been trying to duplicate the problem you reported. In my case, I copied my SF folder from a Krypton system to my Matrix test system. After loading the SF addon, I was able to set the SF directory to the newly copied  SF folder. Everything worked as it should except for certain items that utilized addons that were not compatible with Kodi Matrix.

I just released SF version 2.0.11 that fixed a new issue that was reported. If this version does not work for you, please go into SF settings,  turn Debug On and then post a new log. The previous log you posted did not contain any SF debug entries.

Thanks

**Comment by rizreflects** [comment]
Just tested the new version. The problem still persists for me. I have taken screenshots and will walk you through the entire process so hopefully its easier for you to reproduce.

1- In my case, I am adding a liked list from Seren into Superfavourites. To do this navigate to Seren > my movies > my liked movie lists and select any list you like. 

![1addingfromseren](https://user-images.githubusercontent.com/76912707/131224079-3e8585d9-675d-4778-a5f6-0846056f5163.png)
(Christmas movies was selected)

2- Click onto the superfavourites folder as so:

![2superfavouritesmenu](https://user-images.githubusercontent.com/76912707/131224102-3ed0fff9-5c7c-49cc-a94d-99737a0b1da7.png)

3- Create a new folder inside the superfavourites folder called 'Sub-genres'

![3createsubgenres](https://user-images.githubusercontent.com/76912707/131224117-c0be6819-cbe2-445c-b18f-d1073f1ad0d3.png)

4- Save 'Christmas movies' into the newly created 'Sub-genres' folder

![4savetosubgenres](https://user-images.githubusercontent.com/76912707/131224153-26804d37-e91e-402d-bd41-d3447e98d3d8.png)

5- The next step is to create a submenu directed to the Sub-genres folder created earlier. This is done like so (skin used is called Cosmic)

![5createsubmenu](https://user-images.githubusercontent.com/76912707/131224184-bc1268d0-a4e0-45ee-a8bb-d840f1861d6f.png)

6- As you can see the submenu has now been created

![6submenucreated](https://user-images.githubusercontent.com/76912707/131224203-d8824e19-7931-4ae0-9107-f04fce5bb436.png)

7- This is where the problem occurs. If i click 'Christmas Movies' the list will not open. The loading symbol will last for about a second but the list will never actually open. This has been tested on both windows and CoreELEC.

![christmasmovieswontopen](https://user-images.githubusercontent.com/76912707/131224243-89325f41-1208-4126-8a85-bf84a050b900.png)

I had enabled debugging in superfavourites settings before creating this menu. As soon as the list did not open I uploaded my log file:

https://paste.kodi.tv/ezuwifivoy

**Comment by jmooremcc** [comment]
I know you stated that you turned on Debug logging, but I'm not seeing any SF Debug entries in the log. With Debug logging on you should see entries like this:
      2021-08-29 11:00:24.438 T:10252   DEBUG <general>: Super Favourites V2.0.11 : ------------------ Launch Parameters ------------------
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : launchMode  = video
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : contentType = 
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : viewType    = 0
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : isHome      = False
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : itemIndex   = -1
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : ident       = 8
      2021-08-29 11:00:24.439 T:10252   DEBUG <general>: Super Favourites V2.0.11 : -------------------------------------------------------
      2021-08-29 11:00:24.443 T:10252   DEBUG <general>: Super Favourites V2.0.11 : ***activateWindowCommand called
      2021-08-29 11:00:24.443 T:10252   DEBUG <general>: Super Favourites V2.0.11 : xbmc.executebuiltin(cmd)

A closer look at your log shows the reason why Debug logging is disabled:
    **2021-08-28 16:56:34.057 T:8460     INFO <general>: Disabled debug logging due to GUI setting. Level 0.**

I recommend that you put the following in your advancedsettings.xml file:
          **\<loglevel>1\</loglevel>**

**Comment by rizreflects** [comment]
Sorry for the late reply. I've added the loglevel command to my advanced settings as you said. I also double checked debugging was enabled in superfavourites which it is. Here's the new log, is this one any better?

https://paste.kodi.tv/uxucoxufab

**Comment by jmooremcc** [comment]
Much better. I can work with this log.
Thank you.

**Comment by rizreflects** [comment]
This kodi build is my test build which contains quite a lot of addons.. If you want me to uninstall most of them to clean up the log a little more just let me know.

**Comment by jmooremcc** [comment]
That would be helpful if it's not too much trouble.

**Comment by rizreflects** [comment]
https://paste.kodi.tv/vidipadete

Hopefully this one should be better

**Comment by jmooremcc** [comment]
Since you said version 2.0.7 worked for you, I did a comparison between 2.0.7 and the current version 2.0.11

If  you are comfortable editing python code, I'd like to try an experiment.
I'd like for you to modify the player.py source file which is located here: **addons/plugin.program.super.favourites/resources/player.py**

The section of code you will be editing looks like this:
>
            if "2Fcategories" in cmd:
                utils.log("xbmc.executebuiltin(cmd)")
                xbmc.executebuiltin(cmd)
            elif 'mode=' in plugin.lower():
                utils.log("xbmc.executebuiltin(cmd)")
                xbmc.executebuiltin(cmd)
            # elif pluginArgs:
            #     xbmc.executebuiltin(cmd)
            else:
                utils.log("xbmc.executebuiltin('RunPlugin(%s)' % plugin)")
                xbmc.executebuiltin('RunPlugin(%s)' % plugin)
>

The two lines you will be editing are at line 172 and look like this:
>
          # elif pluginArgs:
          #     xbmc.executebuiltin(cmd)
>
I would like for you to uncomment those 2 statements by removing the # from in front of the 2 statements. Once you do this, the code should look like this:
>
            if "2Fcategories" in cmd:
                utils.log("xbmc.executebuiltin(cmd)")
                xbmc.executebuiltin(cmd)
            elif 'mode=' in plugin.lower():
                utils.log("xbmc.executebuiltin(cmd)")
                xbmc.executebuiltin(cmd)
            elif pluginArgs:
                xbmc.executebuiltin(cmd)
            else:
                utils.log("xbmc.executebuiltin('RunPlugin(%s)' % plugin)")
                xbmc.executebuiltin('RunPlugin(%s)' % plugin)
>

Let me know if this mod fixes the problem you reported.

Thanks

**Comment by rizreflects** [comment]
FYI my player.py file is not in the resources folder but rather addons > plugin.program.super.favourites. The resources folder only contains settings.xml along with more folders. I'm going to make the changes you told me now and get back to you soon.

**Comment by rizreflects** [comment]
I removed the two #'s like you directed.. this is how the code looked after

 ![IMG_20210905_145356.jpg](https://user-images.githubusercontent.com/76912707/132129601-ee0a75c0-2c44-48c3-a43e-6db3d1ba3fc5.jpg)

But now superfavourites crashes as soon as I try to add anything from it (in my case I was trying to add the same submenu as before.. the 'sub-genres' one). When superfavourites crashed it told me to check the log file.. here it is:

https://paste.kodi.tv/umoyinesux

**Comment by jmooremcc** [comment]
The problem you're experiencing is caused by the alignment of the 2 lines of code you modified. If you look closely they are shifted right by one space. Python is very finicky when it comes to indenting text. All you have to do is place your cursor in front of each line's text and hit the backspace key once. That will align the 2 line's properly.

**Comment by rizreflects** [comment]
Sorry about the late reply. I can confirm it's working properly after making the last change.

**Comment by jmooremcc** [comment]
Great. I will do some more testing for side effects and then do a general release.

**Comment by rizreflects** [comment]
Thank you. It'll be great to use your addon once again.
